packages/picasso.js/src/core/scene-graph/display-objects/image.js

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable no-undef */
+import extend from 'extend';
+import DisplayObject from './display-object';
+import { rectToPoints, getMinMax } from '../../geometry/util';
+
+/**
+ * @private
+ * @extends DisplayObject
+ * @typedef {object} ImageNode
+ * @property {number} x - X coordinate
+ * @property {number} y - Y coordinate
+ * @property {number} width - Width
+ * @property {number} height- Height
+ * @property {string} src- Image source
+ * @property {number} cx - Center X coordinate (for circle symbol)
+ * @property {number} cy - Center Y coordinate (for circle symbol)
+ * @property {number} r - Radius (for circle symbol)
+ */
+
+export default class Image extends DisplayObject {
+  constructor(...s) {
+    super('image');
+    this.set(...s);
+  }
+
+  set(v = {}) {
+    const { x = 0, y = 0, width = 0, height = 0, rx = 0, ry = 0, src, collider, cx = 0, cy = 0, r = 0 } = v;
+    let opts;
+    if (v.symbol === 'circle') {
+      opts = extend(
+        {
+          type: 'circle',
+          cx,
+          cy,
+          r,
+        },
+        collider
+      );
+    } else {
+      opts = extend(
+        {
+          type: 'bounds',
+          x,
+          y,
+          width,
+          height,
+        },
+        collider
+      );
+    }
+    super.set(v);
+
+    if (width >= 0) {
+      this.attrs.x = x;
+      this.attrs.width = width;
+    } else {
+      this.attrs.x = x + width;
+      this.attrs.width = -width;
+    }
+
+    if (height >= 0) {
+      this.attrs.y = y;
+      this.attrs.height = height;
+    } else {
+      this.attrs.y = y + height;
+      this.attrs.height = -height;
+    }
+
+    if (rx > 0) {
+      this.attrs.rx = rx;
+    }
+    if (ry > 0) {
+      this.attrs.ry = ry;
+    }
+
+    if (src) {
+      this.attrs.src = src;
+    }
+    this.attrs.r = r;
+    this.attrs.cx = cx;
+    this.attrs.cy = cy;
+    this.collider = opts;
+    this.__boundingRect = { true: null, false: null };
+    this.__bounds = { true: null, false: null };
+    // If no collider is provided, we add an updateCollider function that updates the collider based on the image attributes.
+    if (!v.collider) {
+      this.attrs.updateCollider = this.updateCollider.bind(this);
+    }
+  }
+
+  // This function updates the collider properties based on the image attributes. This function is called in the rendering phase.
+  updateCollider(img) {
```

Comment:
It's been 8 or 9 years since I worked on this code so my memory is failing me here but why do you need to re-create the collider during the rendering phase?